Content.Server/Implants/SubdermalImplantSystem.cs

```diff
@@ -72,6 +86,54 @@ private void OnActivateImplantEvent(EntityUid uid, SubdermalImplantComponent com
         args.Handled = true;
     }
 
+    private void OnScramImplant(EntityUid uid, SubdermalImplantComponent component, UseScramImplantEvent args)
+    {
+        if (component.ImplantedEntity is not { } ent)
+            return;
+
+        if (!TryComp<ScramImplantComponent>(uid, out var implant))
+            return;
+
+        var xform = Transform(ent);
+        var entityCoords = xform.Coordinates.ToMap(EntityManager);
+
+        // try to find a valid position to teleport to, teleport to whatever works if we can't
+        var targetCoords = new MapCoordinates();
+        for (var i = 0; i < implant.TeleportAttempts; i++)
+        {
+            var distance = implant.TeleportRadius * _random.NextFloat(MathF.Sqrt(1f)); // to get an uniform distribution
+            targetCoords = entityCoords.Offset(_random.NextAngle().ToVec() * distance);
+
+            // prefer teleporting to grids
+            if (!_mapManager.TryFindGridAt(targetCoords, out var gridUid, out var grid))
+                continue;
+
+            // the implant user probably does not want to be in your walls
+            var physQuery = GetEntityQuery<PhysicsComponent>();
```

Comment:
store this on system and set in Initialize

Reply:
all other code i saw (many systems) initialises query on the spot

Reply:
bad legacy code noob